code/game/objects/items/devices/scanners/plant_analyzer.dm

```diff
@@ -0,0 +1,275 @@
+/obj/item/plant_analyzer
+	name = "plant analyzer"
+	desc = "A scanner used to evaluate a plant's various areas of growth, genetic traits and chemicals."
+	icon = 'icons/obj/devices/scanner.dmi'
+	icon_state = "hydro"
+	inhand_icon_state = "analyzer"
+	worn_icon_state = "plantanalyzer"
+	lefthand_file = 'icons/mob/inhands/equipment/tools_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/tools_righthand.dmi'
+	w_class = WEIGHT_CLASS_TINY
+	slot_flags = ITEM_SLOT_BELT
+	custom_materials = list(/datum/material/iron = SMALL_MATERIAL_AMOUNT*0.3, /datum/material/glass =SMALL_MATERIAL_AMOUNT*0.2)
+	/// Cached data from ui_interact
+	var/list/last_scan_data
+
+/obj/item/plant_analyzer/Initialize(mapload)
+	. = ..()
+	register_item_context()
+
+/obj/item/plant_analyzer/add_item_context(
+	obj/item/source,
+	list/context,
+	atom/target,
+)
+
+	if(isliving(target))
+		// It's a health analyzer, but for podpeople.
+		var/mob/living/living_target = target
+		if(!(living_target.mob_biotypes & MOB_PLANT))
+			return NONE
+
+		context[SCREENTIP_CONTEXT_LMB] = "Scan health"
+		context[SCREENTIP_CONTEXT_RMB] = "Scan chemicals"
+		return CONTEXTUAL_SCREENTIP_SET
+
+	if(isitem(target))
+		// Easier to handle this here, as grown items are split across two type-paths
+		var/obj/item/item_target = target
+		if(!item_target.get_plant_seed())
+			return NONE
+
+		context[SCREENTIP_CONTEXT_LMB] = "Scan plant stats"
+		return CONTEXTUAL_SCREENTIP_SET
+
+	return NONE
+
+/// When we use the analyzer in hand - try to show the results of the last scan
+/obj/item/plant_analyzer/attack_self(mob/user, modifiers)
+	if(user.stat != CONSCIOUS || !user.can_read(src) || user.is_blind())
+		return
+	if(!last_scan_data)
+		return
+	ui_interact(user)
+
+/// When we attack something, first - try to scan something we hit with left click. Left-clicking uses scans for stats
+/obj/item/plant_analyzer/pre_attack(atom/target, mob/living/user)
```

Comment:
Could swap this to use the interact procs (rather than attack) if you feel so inclined